graylog2-server/src/main/java/org/graylog2/indexer/indices/IndexStatistics.java

```diff
@@ -17,27 +17,134 @@
 package org.graylog2.indexer.indices;
 
 import com.google.auto.value.AutoValue;
-import org.elasticsearch.action.admin.indices.stats.CommonStats;
-import org.elasticsearch.cluster.routing.ShardRouting;
-import org.graylog.autovalue.WithBeanGetter;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.graylog2.indexer.cluster.Cluster;
+import org.graylog2.rest.models.system.indexer.responses.IndexInfo;
+import org.graylog2.rest.models.system.indexer.responses.IndexStats;
+import org.graylog2.rest.models.system.indexer.responses.ShardRouting;
 
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static org.graylog2.indexer.gson.GsonUtils.asBoolean;
+import static org.graylog2.indexer.gson.GsonUtils.asJsonArray;
+import static org.graylog2.indexer.gson.GsonUtils.asJsonObject;
+import static org.graylog2.indexer.gson.GsonUtils.asLong;
+import static org.graylog2.indexer.gson.GsonUtils.asString;
 
 @AutoValue
-@WithBeanGetter
 public abstract class IndexStatistics {
     public abstract String indexName();
 
-    public abstract CommonStats primaries();
+    public abstract JsonObject primaries();
```

Comment:
Going from a structured data type to an unstructured one is a regression imo. I think we should keep the original data types and go the extra mile to extract the values in `buildIndexStats()`.

This also helps us to prevent unnecessary changes in consumers of this structure.